src/allotropy/parsers/revvity_kaleido/kaleido_structure.py

```diff
@@ -0,0 +1,410 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from dataclasses import dataclass
+import logging
+import re
+from typing import Optional
+
+import pandas as pd
+
+from allotropy.allotrope.models.plate_reader_benchling_2023_09_plate_reader import (
+    ScanPositionSettingPlateReader,
+    TransmittedLightSetting,
+)
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.lines_reader import CsvReader
+from allotropy.parsers.revvity_kaleido.kaleido_common_structure import (
+    PLATEMAP_TO_SAMPLE_ROLE_TYPE,
+    SCAN_POSITION_CONVERTION,
+    TRANSMITTED_LIGHT_CONVERTION,
+    WellPosition,
+)
+from allotropy.parsers.utils.values import (
+    assert_not_none,
+    try_float,
+    try_float_or_none,
+)
+
+
+@dataclass
+class BackgroundInfo:
+    experiment_type: str
+
+
+@dataclass(frozen=True)
+class Results:
+    barcode: str
+    results: pd.DataFrame
```

Comment:
Storing the raw dataframe in the this dataclass is an anti-pattern.

You should instead process the data in a create function and store it in the dataclass.

The reason for this is that it makes reasoning about the structure and testing easier.

In this example, you should could, for example, create a two-dim array of wells.